docker/Dockerfile

```diff
@@ -1,75 +1,65 @@
-FROM ubuntu:16.04
+ARG device
+ARG username="jovyan"
+
+FROM ubuntu:18.04 AS base-cpu
+
+FROM nvidia/cuda:10.2-base-ubuntu18.04 AS base-gpu
+
+FROM base-$device AS base
+
+ARG username
```

Comment:
Doesn't this override the default value specified before? (I'm not an expert on Dockerfiles.)